comet-accumulo/src/main/java/org/renci/comet/accumuloops/AccumuloOperationsApiImpl.java

```diff
@@ -271,40 +296,63 @@ private static BatchWriter createBatchWriter(Connector conn, String table) {
      */
     public Map<String[], Value> readOneRow(Connector conn, String tableName, Text rowID, Text colFam, Text colQual, String visibility) throws TableNotFoundException, AccumuloException, AccumuloSecurityException {
         Map<String[], Value> output = new HashMap<>();
-        try {
+        Scanner scanner = null;
+        int count = 0;
+        boolean succeed = false;
+        long beginTime = System.currentTimeMillis();
+        while(!succeed) {
```

Comment:
Shouldn't this loop be based on both retry counts and sleep timer. As with current implementation, we will keep on retrying until we succeed which could result in a infinite loop for invalid read/write token passed.

Retry count and sleep interval should be both configurable.

Reply:
Retry for BAD_AUTHORIZATIONS only happens when following conditions are met: 1) it is BAD_AUTHORIZATIONS exception, 2) the user authentication is what has been set, 3) System.currentTimeMillis() <= beginTime + retryDuration. If the conditions are not met, or if it is other type of exceptions, Exception will be thrown, and Infinite loop won't happen. 
This has been tested and verified. 

Only "retryDuration" is used and retry will stop when retryDuration time has reached. 
"count" is just for logging, I think it would be helpful to fine tune the "retryDuration"  value later by analyzing the logs.

Reply:
okay, understood!

Another question: Isn't step 2 redundant ?? As it will always be true since we are comparing the conn created by the incoming visibility parameter itself.
visibility.equals(conn.securityOperations().getUserAuthorizations(user).toString()

Reply:
If there is other head node happen to use the same user account to access accumulo then the authorization might be overwritten. In that case, conn.securityOperations().getUserAuthorizations(user).toString() will get the overwritten value.   
This shouldnt happen so if happen exception will be thrown immediately.